public/js/graph/react_graph.js

```diff
@@ -261,28 +270,79 @@ var Graph = React.createClass({
     },
 
     nodeMouseEnter: function (event) {
-        var courseID = event.currentTarget.id;
-        var currentNode = this.refs.nodes.refs[courseID];
+        var courseId = event.currentTarget.id;
+        var currentNode = this.refs.nodes.refs[courseId];
         currentNode.focusPrereqs(this);
-
-        // Old hover modal code
-        if ($('.modal').length === 0) {
-            $('.tooltip-group').remove();
-            tooltip.displayTooltip(courseID);
+				
+        this.clearAllTimeouts();
+			
+        var infoBox = this.refs.infobox;
+        //TODO: adjust xPos and yPos
+        var xPos = currentNode.props.JSON.pos[0];
+        var yPos = currentNode.props.JSON.pos[1];
+        var rightSide = xPos > 222;
+        // The tooltip is offset with a 'padding' of 5.
+        if (rightSide) {
+                xPos = parseFloat(xPos) - 65;
+        } else {
+                xPos = parseFloat(xPos) +
+                        parseFloat(currentNode.props.JSON.width) + 5;
         }
+
+        yPos = parseFloat(yPos);
+
+        infoBox.setState({xPos: xPos,
+                          yPos: yPos,
+                          nodeId: courseId,
+                          showInfobox: true});
+        
+        var modal = this.refs.modal;
+        modal.setState({courseId: courseId.substring(0,6)});
     },
 
     nodeMouseLeave: function (event) {
-        var courseID = event.currentTarget.id;
-        var currentNode = this.refs.nodes.refs[courseID];
+        var courseId = event.currentTarget.id;
+        var currentNode = this.refs.nodes.refs[courseId];
         currentNode.unfocusPrereqs(this);
+				
+        var infoBox = this.refs.infobox;
```

Comment:
We should be consistent on the capitalization of `infoBox`. Choose `infoBox` or `infobox`, and stick with it.